sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/DynamicDestinationsHelpers.java

```diff
@@ -279,6 +279,11 @@ static class ConstantTimePartitioningClusteringDestinations<T>
     private final @Nullable ValueProvider<String> jsonTimePartitioning;
     private final @Nullable ValueProvider<String> jsonClustering;
 
+    // Lazily initialized and cached values.
+    private @Nullable String evaluatedPartitioning = null;
```

Comment:
is there a concern during serialization/deserialization if these strings not transient?